backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlQuery.java

```diff
@@ -715,122 +706,126 @@ private Field<Object> jsonGroupBySelect(
               .in(jsonFilterQuery(table, column, tableAlias, subAlias, filter, searchTerms));
     }
 
-    List<SelectConnectByStep> subQuery = new ArrayList<>();
-    List<Field> selectFields = new ArrayList<>();
-    List<Field> groupByFields = new ArrayList<>();
+    Set<Field> aggregationFields = new HashSet<>(); // sum(x), count, etc
+    Set<Field> groupByFields = new HashSet<>(); // name, ref{otherName}, etc
+    Set<Field> nonArraySourceFields = new HashSet<>(); // xo x, name, except those from ref_array
+    List<SelectConnectByStep> refArraySubqueries = new ArrayList<>(); // for the ref_array columns
     for (SelectColumn field : groupBy.getSubselect()) {
       if (COUNT_FIELD.equals(field.getColumn())) {
         if (schema.hasActiveUserRole(VIEWER.toString())) {
-          selectFields.add(field("COUNT(*)"));
+          aggregationFields.add(field("COUNT(*)"));
         } else {
-          selectFields.add(field("GREATEST({0},COUNT(*))", 10L).as(COUNT_FIELD));
+          aggregationFields.add(field("GREATEST({0},COUNT(*))", 10L).as(COUNT_FIELD));
         }
+      } else if (SUM_FIELD.equals(field.getColumn())) {
+        List sumFields = new ArrayList<>();
+        // todo add minimal sum value permission check
```

Comment:
need to handle AGG perm only case ( presume no perm case is handled before this method call ?)

Reply:
there is an agg permission check before this is called. 

This 'todo' is about the fact that when you sum values from only one row then sum equals the individual value. A crude solution might be to disable 'sum' unless you have view permission. A more sophisticated solution would require 'count' to be of a greater value then 5 to show the sum (that actually might not be that hard).

Reply:
Added a crude fix